src/multirust-cli/proxy_mode.rs

```diff
@@ -0,0 +1,28 @@
+use common::{run_inner, set_globals};
+use multirust::{Cfg, Result, Error};
+use multirust_utils::utils;
+use std::env;
+use std::path::PathBuf;
+
+pub fn main() -> Result<()> {
+    try!(::self_update::cleanup_self_updater());
```

Comment:
In most cases all it's going to do is stat a single file. I could also only do this in multirust mode, but I think it'll be uncommon to run `multirust` soon after doing an update.